core/annotations/annotations.go

```diff
@@ -28,16 +29,39 @@ const (
 type ID struct {
 	Kind Kind
 	Name string
+
+	// Parts is optional and is used to store additional information about the
+	// entity.
+	Parts any
```

Comment:
Won't this become a very messy and error-prone bag where we'll put everything and anything we need to find a resource by ID?

Reply:
The alternative is to change the following and provide a switch over the names tag.

https://github.com/juju/juju/blame/76e5a5736287a4dc3173e68451eba7eebd5be6b8/apiserver/facades/client/annotations/annotations.go#L99-L107

So it will become:

```go
func (api *API) getEntityAnnotations(ctx context.Context, entityTag string) (map[string]string, error) {
	tag, err := names.ParseTag(entityTag)
	if err != nil {
		return nil, errors.Trace(err)
	}

	switch tag.Kind() {
	case names.ApplicationTagKind:
		return api.getApplicationAnnotations(ctx, entityTag)
	case names.CharmTagKind:
		return api.getCharmAnnotations(ctx, entityTag)
	case names.MachineTagKind:
		return api.getMachineAnnotations(ctx, entityTag)
	case names.ModelTagKind:
		return api.getModelAnnotations(ctx, entityTag)
	case names.UnitTagKind:
		return api.getUnitAnnotations(ctx, entityTag)
	default:
		return nil, errors.Errorf("unsupported entity kind %q", tag.Kind())
	}
}
```

Reply:
I don't mind that, but it does mean more work.